lib/cocina/models/validators/map_coordinates_validator.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'geo/coord'
+
+module Cocina
+  module Models
+    module Validators
+      # Validates that Purl matches the external identifier (druid)
+      class MapCoordinatesValidator
+        def self.validate(clazz, attributes)
+          new(clazz, attributes).validate
+        end
+
+        def initialize(clazz, attributes)
+          @clazz = clazz
+          @attributes = attributes
+        end
+
+        def validate
+          return unless meets_preconditions?
+
+          # valid condition
+          return if invalid_coordinates.empty?
+
+          raise ValidationError, "Invalid map coordinates for #{druid}."
+        end
+
+        private
+
+        attr_reader :clazz, :attributes
+
+        def meets_preconditions?
+          map_coordinates
+        end
+
+        def invalid_coordinates
+          @invalid_coordinates ||= map_coordinates.select { |subject| invalid?(subject[:value]) }
+        end
+
+        def druid
+          @druid ||= attributes[:externalIdentifier]
+        end
+
+        def map_coordinates
+          @map_coordinates ||= attributes.dig(:description, :subject)&.select { |subject| subject[:type] == 'map coordinates' }
+        end
+
+        # coordinate validation from:
+        # https://github.com/iiif-prezi/osullivan/blob/9816ed3dc2fd739655523620f2602f4494310674/lib/iiif/v3/presentation/nav_place.rb#L17
+        COORD_REGEX = /(?<hemisphere>[NSEW]) (?<degrees>\d+)[°⁰*] ?(?<minutes>\d+)?[ʹ']? ?(?<seconds>\d+)?[ʺ"]?/
+
+        def invalid?(coordinate_text)
+          coordinates(coordinate_text).nil? || coordinates(coordinate_text).empty?
+        end
```

Comment:
it looks like we have `activesupport` as a dependency, so I think that means you can do `coordinates(coordinate_text).blank?` which is true if it's nil or empty.

Reply:
might need to `require 'active_support/core_ext/object/blank'`